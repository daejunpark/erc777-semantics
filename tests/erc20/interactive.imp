test.preamble();

// Interactive program
int continueLoop;
continueLoop = 1;
while (continueLoop == 1) {
  print("--------------------------------\n");
  print("Caller: "); print(caller); print("\n");
  print("--------------------------------\n");
  print(" 1 : totalSupply()\n");
  print(" 2 : balanceOf(Id)\n");
  print(" 3 : allowance(Owner,Spender)\n");
  print(" 4 : approve(Spender,Allowance)\n");
  print(" 5 : transfer(To,Value)\n");
  print(" 6 : transferFrom(From,To,Value)\n");
  print("--------------------------------\n");
  print(" 0 : EXIT\n");
  print("-2 : makeCaller(Id)\n")
  print("--------------------------------\n");
  print("Command: ");
  c = read();
  print("\n\n\n\n\n\n\n");
  if (c == -2) { test.makeCaller(read()); }
  if (c == 0 ) { continueLoop = 0; }
  if (c == 1 ) { test.totalSupply(); }
  if (c == 2 ) { test.balanceOf(read()); }
  if (c == 3 ) { test.allowance(read(), read()); }
  if (c == 4 ) { test.approve(read(), read()); }
  if (c == 5 ) { test.transfer(read(), read()); }
  if (c == 6 ) { test.transferFrom(read(), read(), read()); }
}
