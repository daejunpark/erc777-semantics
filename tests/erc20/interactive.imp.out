--------------------------------
Caller: 7
--------------------------------
 1 : totalSupply()
 2 : balanceOf(Id)
 3 : allowance(Owner,Spender)
 4 : approve(Spender,Allowance)
 5 : transfer(To,Value)
 6 : transferFrom(From,To,Value)
--------------------------------
 0 : EXIT
-2 : makeCaller(Id)
--------------------------------
Command: 






